### PR TITLE
feat(cli): improve store config typehints, prepare for static array support

### PR DESCRIPTION
**Pull request by dk1a**
This mostly just improves typehints, in particular the specific field/key with an invalid type is now highlighted, instead of a vague ts error about enums not having correct keys.
The improvements also allow to easily plug in support for static arrays, so that they just work without defining them as user types (this PR does not actually add support for static arrays)

### CODE REVIEW

File: packages/cli/src/config/parseStoreConfig.ts [hunk 1]
```diff
@@ -131,18 +134,29 @@ export const zTablesConfig = z.record(TableName, zTableConfig).transform((tables
  *
  ************************************************************************/
 
-export type EnumsConfig<EnumNames extends StringForUnion = StringForUnion> = string extends EnumNames
+export type EnumsConfig<EnumNames extends StringForUnion> = never extends EnumNames
```
Comment (holic) [inline packages/cli/src/config/parseStoreConfig.ts:137]: how does this condition work?

File: packages/cli/src/config/parseStoreConfig.ts [hunk 2]
```diff
@@ -172,13 +189,17 @@ export type StoreUserConfig<EnumNames extends StringForUnion = StringForUnion> =
...
+  tables: TablesConfig<StaticUserTypes, StaticUserTypes>;
   /** Path to the file where common user types will be generated and imported from. Default is "Types" */
   userTypesPath?: string;
 };
 
 /** Type helper for defining StoreUserConfig */
-export function storeConfig<EnumNames extends StringForUnion = StringForUnion>(config: StoreUserConfig<EnumNames>) {
+export function storeConfig<
+  // (`never` is overridden by inference, so only the defined enums can be used by default)
+  EnumNames extends StringForUnion = never,
```
Comment (holic) [inline packages/cli/src/config/parseStoreConfig.ts:200]: oh I see here!

File: packages/cli/src/config/validation.ts [hunk 3]
```diff
@@ -151,3 +151,13 @@ export function validateSelector(name: string, ctx: RefinementCtx) {
     });
   }
 }
+
+/** Returns null if the type does not look like a static array, otherwise element and length data */
+export function parseStaticArray(abiType: string) {
+  const matches = abiType.match(/^(\w+)\[(\d+)\]$/);
+  if (matches === null) return null;
```
Comment (holic) [inline packages/cli/src/config/validation.ts:158]: slight preference for `== null` here to catch `undefined` (I know the docs say this should return `null` but just wanna be safe)

File: packages/cli/src/config/parseStoreConfig.ts [hunk 4]
```diff
@@ -172,13 +189,17 @@ export type StoreUserConfig<EnumNames extends StringForUnion = StringForUnion> =
    *  - abi or user type for a single-value table (aka ECS component).
    *  - FullTableConfig object for multi-value tables (or for customizable options).
    */
-  tables: TablesConfig<EnumNames>;
+  tables: TablesConfig<StaticUserTypes, StaticUserTypes>;
```
Comment (holic) [inline packages/cli/src/config/parseStoreConfig.ts:192]: is this change intentional?

**Review by holic — COMMENTED** [review]
just one tiny thing and a question but otherwise looks great!

File: packages/cli/src/config/validation.ts (hunk 3 above)
Comment (dk1a) [reply packages/cli/src/config/validation.ts]: changed it to just `!matches`, might as well catch any falsey value then

File: packages/cli/src/config/parseStoreConfig.ts (hunk 4 above)
Comment (dk1a) [reply packages/cli/src/config/parseStoreConfig.ts:192]: yes, for now it doesn't do much of anything, but the idea is to separate all types for fields, and static types for primary keys; since I'll be adding static user types probably. I imagine I'll be changing these generics again after figuring out how exactly I want to use them

**Review by holic — APPROVED** [review]
